frontend/src/components/Campaign/CampaignTable.tsx

```diff
@@ -1,41 +1,93 @@
 import { ApplicantData } from "@/lib/definitions";
-import { Table, TableCaption, TableContainer, Tbody, Td, Th, Thead, Tr } from "@chakra-ui/react"
-import React from "react";
+import request from "@/util/api";
+import {
+  Table,
+  TableCaption,
+  TableContainer,
+  Tbody,
+  Td,
+  Th,
+  Thead,
+  Tr,
+  useToast,
+} from "@chakra-ui/react";
+import React, { useEffect, useState } from "react";
+import { useQuery } from "react-query";
 
 type CampaignTableProps = {
-    campaignName: string;
-    tableData: ApplicantData[];
-}
+  campaignId: string;
+};
 
-export const CampaignTable: React.FC<CampaignTableProps> = ({ campaignName, tableData }) => {
-    const campaignApplicantData = () => {
-        return tableData.map((data: ApplicantData, index: number) => (
-            <Tr key={index + 1}>
-                <Td>{index + 1}</Td> {/* Needs to be replaced with actual rank from LLM */}
-                <Td>{data.contactDetails.applicantName}</Td> {/* Can either make this a link or add a button to more details */}
-                <Td>13/07/2024</Td>
-                <Td>89%</Td>
-            </Tr>
-        ));
+export const CampaignTable: React.FC<CampaignTableProps> = ({ campaignId }) => {
+  const [campaignName, setCampaignName] = useState("");
+  const [campaignData, setCampaignData] = useState([]);
+  const toast = useToast();
+
+  const { isLoading } = useQuery(
+    [`load${campaignId}`, campaignId],
+    () =>
+      fetch(
+        `${import.meta.env.VITE_BACKEND_URL}/api/campaign/load/${campaignId}`,
+        {
+          credentials: "include",
+        }
+      )
+        .then((res) => {
+          return res.json();
+        })
+        .then((data) => {
+          setCampaignName(data.currentCampaign.documentName);
+          setCampaignData(data.currentCampaign.applicant);
+        })
+        .catch((error) => {
+          console.error("Load campaign error:", error);
+          toast({
+            title: `Load Campaign Error`,
+            description: `Error loading campaign: ${error}`,
+          });
+        }),
+    {
+      refetchInterval: false,
+      enabled: true,
+      refetchOnMount: "always",
+      refetchOnWindowFocus: false,
+      cacheTime: 600,
+      staleTime: 600,
     }
+  );
 
-    return (
-        <TableContainer>
+  return (
+    <>
+      {isLoading ? (
+        <h1>Loading...</h1>
+      ) : (
+        <>
+          <TableContainer>
             <Table variant="simple">
-                <TableCaption>{campaignName} Applicants</TableCaption>
-                <Thead>
-                    <Tr>
-                        <Th>Rank</Th>
-                        <Th>Applicant Name</Th>
-                        <Th>Date Submitted</Th>
-                        <Th>Score</Th>
-                    </Tr>
-                </Thead>
-                <Tbody>
-                    {campaignApplicantData()}
-                </Tbody>
+              <TableCaption>{campaignName} Applicants</TableCaption>
```

Comment:
see how you only use campaignName as a string?  u can use a single campaign object like

const [campaign, setCampaign] 

and the type would be like

{
campaignName: string,
campaignDescription: array or object
}